feat: implement Neo4j integration and graph API routes (BE Only)

avii09 opened the pull request:
Add Neo4j configuration, implement graph data extraction, and create dedicated API routes for graph functionality. Update dependencies to include Neo4j support and enhance testing for the new features.

refers #5

avii09 committed:
update dependencies

avii09 committed:
update dependencies

avii09 committed:
update dependencies

avii09 committed:
update dependencies

avii09 committed:
feat: added neo4j configs

avii09 committed:
feat: added neo4j configs

avii09 committed:
feat: neo4j implementation & entity extraction for graph nodes

avii09 committed:
feat: neo4j implementation & entity extraction for graph nodes

avii09 committed:
feat: neo4j implementation & entity extraction for graph nodes

avii09 committed:
feat: neo4j implementation & entity extraction for graph nodes

avii09 committed:
added graph route

avii09 committed:
feat: added testing for neo4j

avii09 committed:
feat: added testing for neo4j

areebahmeddd commented on backend/app/services/neo4j_service.py:
```diff
@@ -0,0 +1,210 @@
+Handles node creation, relationship management, and graph queries for forensic data.
+Configured for Neo4j Aura (Cloud).
+"""
+
```
1. remove docstring here
2. can you remove this class based approach, can we keep it function based?

areebahmeddd committed:
Delete frontend/package.json

areebahmeddd committed:
Delete frontend/package.json

therealhamad committed:
fix: fixed missing headers in JSON files created during conversion

SpaceTesla committed:
feat: update Navbar with logo and text branding

- Replaced the Home icon with a logo image for enhanced branding.
- Added a text label "cognito.ai" next to the logo for better visibility.

SpaceTesla committed:
style: update Navbar and ThemeToggle component styles

- Adjusted Navbar icon size and text color for better visibility.
- Increased size of Sun and Moon icons in ThemeToggle for improved accessibility.
- Updated button size in ThemeToggle for a more consistent UI experience.

SpaceTesla committed:
feat: add ThemeLogo component and update Navbar

- Introduced a new ThemeLogo component that dynamically switches between light and dark mode logos.
- Replaced the static logo image in the Navbar with the new ThemeLogo component for improved theme support.
- Added a dark mode logo image to enhance branding consistency across themes.

SpaceTesla committed:
style: update ThemeLogo component className

- Removed the default 'rounded-sm' class from the ThemeLogo component for a cleaner design.

areebahmeddd committed:
chore: update versions

areebahmeddd committed:
chore: remove old folder related to TSV (Hamad)

areebahmeddd committed:
feat: integrate TSV -> JSON pipeline

areebahmeddd committed:
fix: update lock file

areebahmeddd commented on backend/app/core/config.py:
```diff
@@ -16,6 +16,12 @@ class Settings(BaseSettings):
     gemini_api_key: Optional[str] = None
 
+    # Neo4j configuration
+    neo4j_uri: str = "bolt://localhost:7687"
```
This requires neo4j to be running locally rather than docker compose way 🤔

areebahmeddd commented on backend/app/routes/graph.py:
```diff
@@ -0,0 +1,161 @@
+from ..models.schemas import UFDRDocument
+
+router = APIRouter(tags=["graph"])
+
```
No need to define tags here in the router, you already defined them in api.py (https://github.com/areebahmeddd/cognito.ai/pull/8/files#diff-0a5ad2428527cd2239e6d7990b9295a7dad823417583c4aa139dd4d6c35d29eaR8)

areebahmeddd commented on backend/app/services/neo4j_service.py:
```diff
@@ -0,0 +1,210 @@
+        """Establish connection to Neo4j Aura database."""
+        try:
+            # For Neo4j Aura (Cloud) - UPDATE THESE WITH YOUR CREDENTIALS
+            uri = "neo4j+s://your-instance.databases.neo4j.io"  # Change this to your Aura URI
```
1. No need for cloud, should be docker way only.

2. also take username password for neo4j from .env and update .env.example with NEO4J_USERNAME, NEO4J_PASSWORD
tip: in core/config.py, within the Config class, we're listening for .env so that's already happening

areebahmeddd commented on frontend/package-lock.json:
this is irrelevant to our backend PR

areebahmeddd commented on frontend/package.json:
irrelevant

areebahmeddd commented:
Btw i've made some commits in the main branch that re structures a few files, so resolving conflicts on your own will be craaazzy, do ping me whenever your free 🙂

Or you can rebase the pr with main and fix em yourself 💀

areebahmeddd committed:
chore: run ruff linter for clean code

areebahmeddd committed:
feat: 6am commit [ need sleep ]
- integrate hamad's new schema

areebahmeddd committed:
fix: default docker network

areebahmeddd committed:
temp: fix query engine

SpaceTesla committed:
feat: add MobileNav component to Navbar

- Introduced a new MobileNav component for improved mobile navigation.
- Integrated MobileNav into the Navbar, ensuring it is only visible on smaller screens.
- Added a GitHub link to the Navbar for easy access to the repository.

SpaceTesla committed:
feat: enhance MobileNav and Navbar components

- Added a GitHub link to the MobileNav for quick access to the repository.
- Changed the SheetContent position in MobileNav from 'left' to 'top' for better layout.
- Updated Navbar to include the ThemeToggle component and adjusted layout for improved responsiveness.

areebahmeddd committed:
Merge main into neo4j-implementation: resolve conflicts
